### PR TITLE
Upg: split notion workflow in 2 and reorg files

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Split notion workflow in 2 workflows and reorg files to simplify code for garbage collection vs sync.

## Risk

Break notion workflows.

## Deploy Plan

Deploy `connector`, restart queues.

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 1]
```diff
@@ -608,47 +599,12 @@ export async function shouldGarbageCollect({
     return false;
   }
 
-  if (
+  // If we are currently doing a full resync, we should not garbage collect
+  const isDoingAFullResync =
```
Comment (fontanierh) [inline connectors/src/connectors/notion/temporal/activities.ts:603]: nit: this is a slight misnomer as we could also be doing the intial sync (which isn't a "re" sync), correct ?

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/activities.ts:603]: Maybe we should rename this entire activity into "isFullSyncOngoing" ? or something (not strongly held though)

File: connectors/src/connectors/notion/temporal/config.ts [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+export const MAX_PENDING_UPSERT_ACTIVITIES = 5;
+
+export const MAX_PENDING_GARBAGE_COLLECTION_ACTIVITIES = 1;
+
+// If set to true, the workflow will process all discovered resources until empty.
+export const PROCESS_ALL_DISCOVERED_RESOURCES = false;
+
+export const DATABASE_TO_CSV_MAX_SIZE = 256 * 1024 * 1024; // 256MB
+
+// the garbageCollect function will be stopped if it runs longer than this
```
Comment (fontanierh) [inline connectors/src/connectors/notion/temporal/config.ts:23]: We could decide to get rid of this as well. WDYT ?

File: connectors/src/connectors/notion/temporal/upserts.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+const {
+  garbageCollectorMarkAsSeenAndReturnNewEntities,
+  fetchDatabaseChildPages,
+  upsertDatabaseStructuredDataFromCache,
+} = proxyActivities<typeof activities>({
+  startToCloseTimeout: "10 minute",
+});
+
+export async function upsertDatabase({
```
Comment (fontanierh) [inline connectors/src/connectors/notion/temporal/upserts.ts:25]: This doesn't need to be exported right ?

**Review by fontanierh — APPROVED** [review]
this is looking great, thanks for cleaning up

File: connectors/src/connectors/notion/temporal/activities.ts (hunk 1 above)
Comment (Fraggle) [reply connectors/src/connectors/notion/temporal/activities.ts:603]: > Maybe we should rename this entire activity into "isFullSyncOngoing" ? or something (not strongly held though)

Yes, now that it's only called by garbage collection workflow, it makes total sense.

File: connectors/src/connectors/notion/temporal/client.ts [hunk 4]
```diff
@@ -42,7 +45,7 @@ export async function launchNotionSyncWorkflow(
     );
   }
 
-  const workflow = await getNotionWorkflow(connectorId, "never");
+  const workflow = await getNotionWorkflow(connectorId, false);
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply connectors/src/connectors/notion/temporal/client.ts:48]: of course !

[the remark this answers is not in the document]
Comment (Fraggle) [reply connectors/src/connectors/notion/temporal/client.ts:48]: You know what, i'll make 2 functions to be super explicit.

File: connectors/src/connectors/notion/temporal/config.ts (hunk 2 above)
Comment (Fraggle) [reply connectors/src/connectors/notion/temporal/config.ts:23]: It is indeed never reached now.

File: connectors/src/connectors/notion/temporal/upserts.ts (hunk 3 above)
Comment (Fraggle) [reply connectors/src/connectors/notion/temporal/upserts.ts:25]: It's called from 2 files (child & admin workflows :()

**Pull request merged**